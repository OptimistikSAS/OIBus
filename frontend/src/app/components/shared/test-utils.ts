import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
