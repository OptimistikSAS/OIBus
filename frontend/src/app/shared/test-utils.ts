import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
