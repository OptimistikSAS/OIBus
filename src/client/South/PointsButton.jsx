import React from 'react'
import PropTypes from 'prop-types'
import { Button } from 'reactstrap'
import { useHistory } from 'react-router-dom'
import ProtocolSchemas from './Protocols.jsx'

const PointsButton = ({ dataSource }) => {
  const history = useHistory()
  const handleEditPoints = () => {
    const link = `/south/${dataSource.dataSourceId}/points`
    history.push({ pathname: link })
  }
  const { points, dataSourceId, protocol, enabled } = dataSource
  const hasButton = ProtocolSchemas[protocol]?.points !== null
  return hasButton ? (
    <Button
      className="inline-button autosize oi-points-button"
      // eslint-disable-next-line no-nested-ternary
      color={enabled ? (points?.length ? 'success' : 'warning') : 'secondary'}
      onClick={() => handleEditPoints(dataSourceId)}
      size="sm"
      outline
    >
      {`Points ${points ? `(${points.length})` : '(0)'}`}
    </Button>
  ) : (
    <Button
      className="inline-button autosize oi-points-button"
      size="sm"
      color="secondary"
      disabled
    >
      File
    </Button>
  )
}

PointsButton.propTypes = { dataSource: PropTypes.object.isRequired }

export default PointsButton
