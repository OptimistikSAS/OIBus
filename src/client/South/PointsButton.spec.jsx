import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import testConfig from '../../../tests/testConfig'
import PointsButton from './PointsButton.jsx'

jest.mock('react-router-dom', () => (
  { useHistory: jest.fn().mockReturnValue([]) }
))

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

const dataSource = testConfig.south.dataSources[0]

describe('PointsButton', () => {
  test('check PointsButton', () => {
    act(() => {
      ReactDOM.render(
        <PointsButton
          dataSource={dataSource}
        />, container,
      )
    })
    expect(container).toMatchSnapshot()
  })
})
