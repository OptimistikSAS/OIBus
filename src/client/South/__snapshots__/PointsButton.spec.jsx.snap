// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PointsButton check PointsButton 1`] = `
<div>
  <button
    class="inline-button autosize oi-points-button btn btn-outline-secondary btn-sm"
    type="button"
  >
    Points (2)
  </button>
</div>
`;
